### ISSUE TITLE
Add src/dst host filter to pscheduler schedule or/and make it grep friendly

### ISSUE THREAD

**Issue by szymontrocha** [perfsonar/pscheduler#1305]
Hi,

Here is an idea for enhancements

When troubleshooting tasks it would be great to have a possibility to filter "pschduler schedule" not only by test type but also by dest or source node.
Other option is also to make the output of "pscheduler" schedule more grep friendly. So far these are three seperate lines:
2022-08-30T13:07:07Z - 2022-08-30T13:07:46Z (Pending)
throughput --source psmp-gn-bw-vie-at.geant.org --source-node psmp-gn-bw-vie-at.geant.org --dest perfsonar.lumii.lv --dest-node perfsonar.lumii.lv --duration PT20S --ip-version 4 --omit PT10S --reverse (Run with tool 'iperf3')
https://psmp-gn-bw-vie-at.geant.org/pscheduler/tasks/79177132-f687-44e4-8516-7557da624c9d/runs/2a7135ae-7ba9-4ce2-9c66-9e8949f61411
so when using grep to filter e.g. hostname one cannnot see the whole entry but just the task specification line. That makes troubleshooting more difficult. 
It would be great to have just a single line antry like (or similar) instead of three so that it's easily processed by grep or similar tools:
2022-08-30T13:07:07Z - 2022-08-30T13:07:46Z (Pending) throughput --source psmp-gn-bw-vie-at.geant.org --source-node psmp-gn-bw-vie-at.geant.org --dest perfsonar.lumii.lv --dest-node perfsonar.lumii.lv --duration PT20S --ip-version 4 --omit PT10S --reverse (Run with tool 'iperf3') [ https://psmp-gn-bw-vie-at.geant.org/pscheduler/tasks/79177132-f687-44e4-8516-7557da624c9d/runs/2a7135ae-7ba9-4ce2-9c66-9e8949f61411 ]

**Comment by mfeit-internet2** [comment]
The ideal way to do this would be to query the API directly and process the output to find the things you want, e.g.:

```
https://localhost/pscheduler/schedule?start=2022-09-21T11:53:02Z&end=2022-09-21T17:53:02Z \
    | jq ' ... Filter ...'
```

#1253 (to be released with 5.0) added a `--format` switch that will let you spit out the raw JSON so you can still use the command line if you want, but that doesn't get the easy `grep` you're after.  I've added intelligence to the program that disables line wrapping of the standard output isn't a TTY.  Ergo:

Normal behavior, stdout is a TTY:
```
$ pscheduler schedule -PT1H
2022-09-21T17:59:39Z - 2022-09-21T17:59:47Z (Finished)
throughput --dest localhost --duration PT5S --link-rtt PT0.00004S --bandwidth
  100000000000 --mss 9000 --ip-version 4 --zero-copy --reverse --loopback (Run
  with tool 'iperf3')
https://localhost/pscheduler/tasks/893a3fa3-1b92-4c76-b3d0-ab7652e6befe/runs/7f004700-23fa-4e9a-b531-43142441f764
```

Standard output forced into a pipe, which isn't a TTY:
```
$ pscheduler schedule -PT1H | cat  
2022-09-21T17:59:39Z - 2022-09-21T17:59:47Z (Finished)
throughput --dest localhost --duration PT5S --link-rtt PT0.00004S --bandwidth 100000000000 --mss 9000 --ip-version 4 --zero-copy --reverse --loopback (Run with tool 'iperf3')
https://localhost/pscheduler/tasks/893a3fa3-1b92-4c76-b3d0-ab7652e6befe/runs/7f004700-23fa-4e9a-b531-43142441f764
```

Same, but clever things with `grep` to grab the context:

```
$ pscheduler schedule -PT12H | fgrep -B 2 -A 2 -- '--dest localhost'

2022-09-21T17:59:39Z - 2022-09-21T17:59:47Z (Finished)
throughput --dest localhost --duration PT5S --link-rtt PT0.00004S --bandwidth 100000000000 --mss 9000 --ip-version 4 --zero-copy --reverse --loopback (Run with tool 'iperf3')
https://localhost/pscheduler/tasks/893a3fa3-1b92-4c76-b3d0-ab7652e6befe/runs/7f004700-23fa-4e9a-b531-43142441f764
```

**Comment by mfeit-internet2** [comment]
**Release Notes:**

Disabled line wrapping in the `schedule` command when standard output is not a TTY to enable use with `grep` and other filters.
